<?xml version="1.0" ?>
<gazebo version="1.2">
  <world name="default">
    <physics type="ode">
      <!-- Gravity default -->	
      <gravity>0 0 -9.81</gravity>
      <update_rate>1000</update_rate>
      <ode>
        <solver>
          <type>quick</type>
          <dt>0.001</dt>
          <iters>50</iters>
          <sor>1.4</sor>
        </solver>
        <constraints>
          <cfm>0.0</cfm>
          <erp>0.2</erp>
          <contact_max_correcting_vel>100.0</contact_max_correcting_vel>
          <contact_surface_layer>0.0</contact_surface_layer>
        </constraints>
      </ode>
    </physics>

    <!-- A global light source -->
    <include>
      <uri>model://sun</uri>
    </include>

    <include>
      <uri>model://ground_plane</uri>
    </include>

    <!-- drchubo -->		
    <include>
      <uri>model://drchubo</uri>
      <pose>1.33 -1.1 1.0 0 0 1.57</pose>
      <plugin filename="libdrchuboPlugin.so" name="drchuboPlugin"/>	
    </include>

    <!-- golf_cart -->
<!--    <include>
      <uri>model://golf_cart</uri>
      <pose>0.5 0.8 0.0 0.0 0.0 0.0</pose>
      <name>vehicle</name>  
    </include> -->

    <!-- drc_vehicle -->
    <include>
      <uri>model://drc_vehicle</uri> 
      <pose>0.5 0.8 0.0 0.0 0.0 0.0</pose>
    </include>


    <!-- world Plugin -->
    <plugin filename="libworldSimPlugin.so" name="worldSimPlugin">
        <robot>
            <model_name>drchubo</model_name>
            <pin_link>Body_Torso</pin_link> 
        </robot>   	
        <vehicle>
            <model_name>drc_vehicle</model_name>
            <seat_link>polaris_ranger_ev::chassis</seat_link> 
        </vehicle>   	
    </plugin>

  </world>
</gazebo>
